tests/unit/dataactvalidator/test_b9_award_financial.py

```diff
@@ -18,126 +19,155 @@ def test_success(database):
     """ Testing valid program activity name for the corresponding TAS/TAFS as defined in Section 82 of OMB Circular
     A-11. """
 
-    af_1 = AwardFinancialFactory(row_number=1, agency_identifier='test', main_account_code='test',
+    af_1 = AwardFinancialFactory(row_number=1, agency_identifier='test', submission_id='1', main_account_code='test',
                                  program_activity_name='test', program_activity_code='test')
 
-    af_2 = AwardFinancialFactory(row_number=2, agency_identifier='test', main_account_code='test',
+    af_2 = AwardFinancialFactory(row_number=2, agency_identifier='test', submission_id='1',  main_account_code='test',
                                  program_activity_name='test', program_activity_code='test')
 
-    pa = ProgramActivityFactory(budget_year=2016, agency_id='test', allocation_transfer_id='test',
+    pa = ProgramActivityFactory(fiscal_year_quarter='FY2017Q1', agency_id='test', allocation_transfer_id='test',
                                 account_number='test', program_activity_name='test', program_activity_code='test')
 
-    assert number_of_errors(_FILE, database, models=[af_1, af_2, pa]) == 0
+    submission = SubmissionFactory(submission_id='1', reporting_fiscal_year='2017', reporting_fiscal_period=3)
```

Comment:
The publish status for each submission should be explicitly set for all tests now that it's in the SQL because we don't want it to accidentally set it to a random number that breaks the test. It's also a more solid test if we know exactly what it is.